### PR TITLE
DUPLO-13021 Support for GCP SQL data source

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Implemented Data Source for GCP SQL
## Summary of changes
DataSource for GCP SQL and SQL list
This PR does the following:

- Implemented `duplocloud_gcp_sql_database_instance` data source
- Implemented `duplocloud_gcp_sql_database_instances` data source

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [✓ ] Manually, on a remote test system

## Describe any breaking changes

- ...

___

## **Type**
enhancement

___

## **Description**
- Added two new data sources for GCP SQL in the DuploCloud provider: `duplocloud_gcp_sql_database_instance` for a single instance and `duplocloud_gcp_sql_database_instances` for a list of instances.
- Implemented Go functions for reading and listing GCP SQL database instances.
- Updated the DuploCloud provider to register the new data sources.
- Added SDK support for listing GCP SQL database instances.
- Updated documentation and examples for the new data sources.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>data_source_duplo_gcp_sql.go</strong><dd><code>Add Data Source for GCP SQL Database Instance</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/data_source_duplo_gcp_sql.go

<li>Added a new data source <code>dataSourceGCPCloudSQL</code> for retrieving a GCP SQL <br>instance in Duplo.<br> <li> Implemented <code>dataSourceGCPSQLCloudRead</code> function for reading GCP SQL <br>instance details.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/433/files#diff-74107f8eef7f8bb26a294f40de4b3c88607e3c48c8b97c5e4a8edebc0bf7b24c">+109/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>data_source_duplo_gcp_sqls.go</strong><dd><code>Add Data Source for GCP SQL Database Instances List</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/data_source_duplo_gcp_sqls.go

<li>Added a new data source <code>dataSourceGCPCloudSQLs</code> for retrieving a list <br>of GCP SQL instances in Duplo.<br> <li> Implemented <code>dataSourceGCPSQLCloudList</code> function for listing GCP SQL <br>database instances.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/433/files#diff-264560f5780092e751dd7b079c3858c46726ac5d3dc1431167f1266f915e098b">+116/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gcp_sql_database.go</strong><dd><code>SDK Support for Listing GCP SQL Database Instances</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplosdk/gcp_sql_database.go

<li>Added <code>GCPSqlDBInstanceList</code> function to list GCP SQL database <br>instances.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/433/files#diff-ef4280fb2cddb49187c078fc9a79d10c1b8a94e27286a2e4c718da2d320ac2d8">+10/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>provider.go</strong><dd><code>Register New GCP SQL Data Sources in Provider</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/provider.go

<li>Registered <code>duplocloud_gcp_sql_database_instance</code> and <br><code>duplocloud_gcp_sql_database_instances</code> data sources in the DuploCloud <br>provider.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/433/files#diff-0959744faba70cfc67abd70733e39a6f668d41c9fae69a27968841ef8ed56eeb">+53/-51</a>&nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Documentation
</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update Changelog for GCP SQL Data Source Support</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
CHANGELOG.md

<li>Updated CHANGELOG.md to include entries for the addition of GCP SQL <br>data source and SQL list data source.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/433/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gcp_sql_database_instance.md</strong><dd><code>Document GCP SQL Database Instance Data Source</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
docs/data-sources/gcp_sql_database_instance.md

<li>Added documentation for the <code>duplocloud_gcp_sql_database_instance</code> data <br>source.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/433/files#diff-98c07f4964f84e272d74f1c993a3eb73ad99f59461a6986ca352d03b765b4264">+32/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gcp_sql_database_instances.md</strong><dd><code>Document GCP SQL Database Instances Data Source</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
docs/data-sources/gcp_sql_database_instances.md

<li>Added documentation for the <code>duplocloud_gcp_sql_database_instances</code> data <br>source.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/433/files#diff-bc69378ec33f3580aa3663d9634ae19da8ecbd3f4037958a37f8c1396d228bdc">+37/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>data-source.tf</strong><dd><code>Example Configuration for GCP SQL Database Instance Data Source</code></dd></summary>
<hr>
      
examples/data-sources/duplocloud_gcp_sql/data-source.tf

<li>Added an example Terraform configuration for the <br><code>duplocloud_gcp_sql_database_instance</code> data source.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/433/files#diff-5e0f482f60c9e763c747eb63144a97976e475eb9e7920969f1c41fbe65aaf7d4">+18/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>data-source.tf</strong><dd><code>Example Configuration for GCP SQL Database Instances Data Source</code></dd></summary>
<hr>
      
examples/data-sources/duplocloud_gcp_sqls/data-source.tf

<li>Added an example Terraform configuration for the <br><code>duplocloud_gcp_sql_database_instances</code> data source.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/433/files#diff-39978ab10934b71184364363d119915d9def083f9c4e76b1aa04911d22494f4d">+10/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: duplocloud/data_source_duplo_gcp_sqls.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceGCPCloudSQLs() *schema.Resource {
+	return &schema.Resource{
+		Description: "`duplocloud_gcp_sql_cloud` retrieves a gcp sql instance in Duplo.",
+
+		ReadContext: dataSourceGCPSQLCloudList,
+
+		Schema: map[string]*schema.Schema{
```
Comment (tahir-duplo) [inline duplocloud/data_source_duplo_gcp_sqls.go:19]: Use same schema structure for resource and data source.

File: duplocloud/data_source_duplo_gcp_sql.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceGCPCloudSQL() *schema.Resource {
+	return &schema.Resource{
+		Description: "`duplocloud_gcp_sql_cloud` retrieves a gcp sql instance in Duplo.",
+
+		ReadContext: dataSourceGCPSQLCloudRead,
+
+		Schema: map[string]*schema.Schema{
```
Comment (tahir-duplo) [inline duplocloud/data_source_duplo_gcp_sql.go:20]: Use same schema structure for resource and data source.

File: Makefile [hunk 3]
```diff
@@ -8,7 +8,7 @@ NAMESPACE=duplocloud
 
 NAME=duplocloud
 BINARY=terraform-provider-${NAME}
-VERSION=0.10.12
+VERSION=0.10.11
```
Comment (tahir-duplo) [inline Makefile:11]: Remove makefile changes.

Comment (Duplo-nikv) [reply Makefile:11]: ok

**Comment by sp1676** [comment]
/update_changelog --pr_update_changelog.push_changelog_changes=true

File: duplocloud/data_source_duplo_gcp_sqls.go (hunk 1 above)
Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_sqls.go]: done

File: duplocloud/data_source_duplo_gcp_sql.go (hunk 2 above)
Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_sql.go]: done

**Comment by Duplo-nikv** [comment]
Closing this pull request this pr has been combined with nodepool datasource pr

**Pull request closed without merging**